src/machine/asm/mod.rs

```diff
@@ -54,6 +54,24 @@ impl CoreMachine for Box<AsmCoreMachine> {
     }
 }
 
+fn inited_memory(machine: &mut AsmCoreMachine, addr: u64, size: u64) -> Result<(), Error> {
+    let frame_from = addr >> MEMORY_FRAME_SHIFTS;
+    let (addr_to, overflowed) = addr.overflowing_add(size);
+    if overflowed {
+        return Err(Error::OutOfBound);
+    }
+    let frame_to = addr_to >> MEMORY_FRAME_SHIFTS;
```

Comment:
Should `frame_to` be `(addr_to - 1) >> MEMORY_FRAME_SHIFTS`? `frame_to` is inclusive below.

Reply:
Yes, a bug here